crates/fuel-core/src/service/adapters/sync.rs

```diff
@@ -0,0 +1,70 @@
+use super::P2PAdapter;
+use fuel_core_services::stream::BoxStream;
+use fuel_core_sync::ports::PeerToPeerPort;
+use fuel_core_types::{
+    blockchain::{
+        primitives::{
+            BlockHeight,
+            BlockId,
+        },
+        SealedBlockHeader,
+    },
+    fuel_tx::Transaction,
+    services::p2p::SourcePeer,
+};
+
+#[cfg(not(feature = "p2p"))]
```

Comment:
Is there a reason to explicitly implement this when `p2p` is disabled rather than just using something like `automock`?

Reply:
automock is a testing only framework. It should never be used in production code

Reply:
I had assumed this impl would mainly be for testing, but would it actually see use? Or rather does there need to be an implementation of `PeerToPeerPort` if `p2p` is disabled, or is this being chosen over attaching feature flags to every function which would call this port?

Reply:
Basically you can't use sync without p2p however there's nothing in fuel sync that depends on p2p so it would be a waste to put it behind a feature flag. Maybe this isn't needed though as we could make the actual sync service optional and only set it when the p2p flag is set

Reply:
Yeh just looked at this and it would require propagating the feature flag to a lot more places. Maybe it's cleaner as is?

Reply:
Making sync optional depending on p2p is preferable for me since it would also reduce compile times for forc testing if fuel-sync doesn't need to be compiled to run local testnets (and forms code clarity perspective) , but I could see that being a separate issue and pr if it's too much. Do you want to do it in this PR, I can approve if not?

Reply:
I don't think it should be an optional dependency. I meant option type like how the maybe relayer works